src/DiabetesApp.tsx

```diff
@@ -1,33 +1,33 @@
 import React from "react";
 import { StyleSheet, Text, View } from "react-native";
-import { createNativeStackNavigator } from "@react-navigation/native-stack";
 
 //Screens
 import AuthScreen from "./screens/authentication/AuthScreen";
 
 // Hooks
 import useAuth from "./hooks/useAuth";
 
-// Costume components
+// Custom components
 import BottomNavigation from "./components/BottomNavigation";
-
-const Stack = createNativeStackNavigator();
+import ErrorModal from "./components/ErrorModal";
 
 const DiabetesApp = () => {
-  const { isInitialized, user, isAuthenticated } = useAuth();
+  const { isInitialized, isAuthenticated } = useAuth();
 
   if (!isInitialized) {
     return (
       <View style={styles.container}>
         <Text>Loading....</Text>
       </View>
     );
-  }
-
-  if (isAuthenticated && user) {
-    return <BottomNavigation />;
   } else {
-    return <AuthScreen />;
+    return (
+      <React.Fragment>
+        {isAuthenticated && <BottomNavigation />}
+        {!isAuthenticated && <AuthScreen />}
+        <ErrorModal />
```

Comment:
Error Modal is always added in the main application since it will appear **above** any screen in order to display the error message. We handle whether the modal should be displayed or not inside the actual `ErrorModal` component.

Reply:
Also in case you guys want more context on what `React.Fragment` is you can check the [official docs](https://reactjs.org/docs/fragments.html). Essentially in react you need to group components to return one big one. 

We could've alternatively wrap this with a `View` component, but it seems redundant since both the `BottomNavigation` and the `AuthScreen` are already wrapped with a `View` inside their respective files. In web based React, you could wrap the whole thing in just a `div` tag as well. All of them achieve the same result, which allows for you to group components (aka JSX elements).